src/gettext.js

```diff
@@ -83,13 +88,85 @@ function ngettext(string, stringPlural, number, replacements, locale_parameter)
         pluralForms = catalogs[locale]["plural-forms"];
         messages = catalogs[locale].messages[string];
     }
-    if (pluralForms && messages && messages.length > 0) {
+    if (pluralForms && messages) {
         if (!pluralFormsFunctions[locale]) {
             pluralFormsFunctions[locale] = helpers.generatePluralFormsFunction(pluralForms);
         }
-        var pluralIndex = pluralFormsFunctions[locale](number);
-        if (messages[pluralIndex] && messages[pluralIndex] !== "") {
-            result = messages[pluralIndex];
+        var pluralIndex;
+        if (messages instanceof Array && messages.length > 0) {
+            pluralIndex = pluralFormsFunctions[locale](number);
+            if (messages[pluralIndex] && messages[pluralIndex] !== "") {
+                result = messages[pluralIndex];
+            }
+        }
+        if (messages["*"] instanceof Array && messages["*"].length > 0) {
+            pluralIndex = pluralFormsFunctions[locale](number);
```

Comment:
To avoid the duplication of plural forms checks, you may organize the code this way:

```
var message;

if (messages instanceof Array ....) {
   message = messages
}

if (messages[*] instanceof Array ....) {
   message = messages[*]
}

if (message[pluralIndex]) {
 result = message[pluralIndex]
}
```

Reply:
I don't get your factorization: in case of format with context, the last `if` will not work.

Reply:
done with [ee2582a](https://github.com/flozz/stone.js/pull/33/commits/ee2582a8ed53a5767f00cc8ee0aca243504ae1bf)

@jbghoul can you confirm that is what you had in mind.

Reply:
yes I confirm

in other words:
* first step: pick message regarding context : `message = findContext(messages)`
* second step : pick plural form: `result = findPlural(message)`